packages/go/trynoports/app_update.go

```diff
@@ -0,0 +1,168 @@
+	cmds = append(cmds, cmd)
+
+	// Recompute the size of the components based on current changes
+	m, cmd = m.ResizeComponents(msg)
+	cmds = append(cmds, cmd)
+
+	// Batch all of the potential commands
+	return m, tea.Batch(cmds...)
+}
+
+// Handle keyboard input
+func (p appState) KeyMsg(msg tea.KeyMsg) (m appState) {
+	m = p
+	switch msg.String() {
+	case "enter":
+		if m.viewport.isRunning || m.list.model.Index() == m.viewport.contentIndex {
+			return
+		}
+		m.viewport.isRunning = true
+		m.viewport.contentIndex = m.list.model.Index()
+
+		item := m.list.model.SelectedItem()
+		var command, args string
+
+		switch item.(type) {
+		case appCommand:
+			// Update the command entry
+			command = item.(appCommand).command
+			args = strings.Join(m.list.model.SelectedItem().(appCommand).args, " ")
+			// Throw away the old contents, otherwise someone malicious might try to allocate infinite memory to that struct
+			m.viewport.content = fmt.Sprintf("> %s %s\n", command, args)
+
+		default:
+			m.viewport.content = ""
+		}
+
+		done := make(chan int, 1)
+		ch, err := m.list.model.SelectedItem().(RunnerWithDone).Run(done)
+		if err != nil {
+			close(done)
+			// Welcome command should never return an error, so it is safe to assume we have an appCommand in here
+			// Thus command & args are set
+			m.viewport.content = fmt.Sprintf("> %s %s\nOops! We messed up, can't run this command right now...", command, args)
+			m.viewport.model.SetContent(m.viewport.content)
+			m.viewport.model.GotoBottom()
+			m.viewport.isRunning = false
+			log.Error("Error running command: ", "error", err)
+			return
+		}
+
+		go func() {
+			for {
+				select {
+				case <-done:
+					close(ch)
+					close(done)
+					m.program.Send(CmdDoneMsg{})
+					return
+				case msg := <-ch:
+					m.program.Send(ViewportContentMsg{msg})
+				}
+			}
+		}()
+	}
+
+	return
+}
+
+// Update the window size when it is reported to us
+func (m appState) ResizeComponents(msg tea.Msg) (appState, tea.Cmd) {
+	// get frame dimensions
+	frameW, frameH := m.frame.style.GetFrameSize()
+
+	// get preferred list dimensions
+	m.list.model.SetSize(80, m.height-frameH)
+
+	// do all the calculations to resize the viewport
+	if m.viewport.isReady {
+		// pseudo render the list to compute it's actual width
+		// (can change during filtering or popping open the full help dialog)
+		listW := lipgloss.Width(m.list.model.View())
+
+		// get list & viewport frame sizes
+		listFrameW, _ := m.list.style.GetFrameSize()
+		viewportFrameW, viewportFrameH := m.viewport.style.GetFrameSize()
+
+		// calculate viewport dimensions
+		viewportW := m.width - frameW - viewportFrameW - listFrameW - listW
+		viewportH := m.height - frameH - viewportFrameH
+
+		// resize the viewport
+		m.viewport.model.Width = viewportW
+		m.viewport.model.Height = viewportH
+
+		content := lipgloss.NewStyle().Width(viewportW).Render(m.viewport.content)
+		m.viewport.model.SetContent(content)
```

Comment:
After resizing the viewport, normalize the content so that it wraps properly in the terminal